### ISSUE TITLE
Select project group at startup

### ISSUE THREAD

**Issue by zeleznypa** [apache/netbeans#4626]
### Description

Netbeans support so called „project groups“ which is very handy, but when I open NetBeans with not preferred project group, it took so many time to switch to another.

### Use case/motivation

It will be very useful to have a possibility to choose the project group before whole NetBeans start.

So when the user have more than one project group defined, the dialog window

<img width="603" alt="Snímek obrazovky 2022-09-12 v 8 12 31" src="https://user-images.githubusercontent.com/1191472/189585219-ee5b34e6-e322-48c2-aceb-61103c846984.png">

should be displayed before projects are loaded (in the startup phase).

Or when somebody run the NetBeans with some argument like `--manage-group`

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by Chris2011** [comment]
I would prefer not to show the same dialog for managing the groups I more prefer to show a list of available group and choose one. The rest is not necessary on startup.

**Comment by zeleznypa** [comment]
@Chris2011 Any possibility to choose the project group will be fine. Reuse dialog window that already exists can be (hopefully) less work, but it is an implementation detail.

**Comment by mbien** [comment]
i think you could start NB with `--close-group` and then select a new one via right click, no?

**Comment by neilcsmith-net** [comment]
> when I open NetBeans with not preferred project group, it took so many time to switch to another.

There is a related issue here that changing groups while projects are loading doesn't work. Making that work correctly and quickly might solve half the issue.

**Comment by zeleznypa** [comment]
@mbien Thanks for that tip. I have no idea, that it exists. I have one step closer to ideal :)

**Comment by RohdeBSH** [comment]
I don’t know why this issue is still open.
This has been possible since at least Netbeans 12.

- `netbeans64.exe --list-groups` ⇒ Lists all available project groups
- `netbeans64.exe --open-group NAME` ⇒ Opens Netbeans with this project group
- `netbeans64.exe --help` ⇒ also displays this in the command line.

**Comment by Chris2011** [comment]
> I don’t know why this issue is still open. This has been possible since at least Netbeans 12.
> 
> * `netbeans64.exe --list-groups` ⇒ Lists all available project groups
> * `netbeans64.exe --open-group NAME` ⇒ Opens Netbeans with this project group
> * `netbeans64.exe --help` ⇒ also displays this in the command line.

I think it is about to choose the group via UI. I'm not opening NetBeans via cli or change the link to my executable everytime I need to change the group at startup. It should be part of the startup process. Show a UI, if you want (option based) reuse opened projects or open an other project group, also change LaF etc.

**Comment by neilcsmith-net** [comment]
Well, if you set the dashboard to always show on startup, you've got easy one-click access to project groups or to switch between light and dark LAFs.  If we expanded the widget list, you could have one just for group switching.

**Comment by RohdeBSH** [comment]
> > I don’t know why this issue is still open. This has been possible since at least Netbeans 12.
> > 
> > * `netbeans64.exe --list-groups` ⇒ Lists all available project groups
> > * `netbeans64.exe --open-group NAME` ⇒ Opens Netbeans with this project group
> > * `netbeans64.exe --help` ⇒ also displays this in the command line.
> 
> I think it is about to choose the group via UI. I'm not opening NetBeans via cli or change the link to my executable everytime I need to change the group at startup. It should be part of the startup process. Show a UI, if you want (option based) reuse opened projects or open an other project group, also change LaF etc.

I don’t open NetBeans via the command line either. For now, I’ve simply created a Windows Start menu entry for the project group and passed the parameter `--open-group NAME` to the .exe file.
Keep it simple...

**Comment by Chris2011** [comment]
> I don’t open NetBeans via the command line either. For now, I’ve simply created a Windows Start menu entry for the project group and passed the parameter `--open-group NAME` to the .exe file. Keep it simple...

Custom windows start menu entry, custom link on the desktop, doesn't matter. This is what I said, you need to configure it with the specific group that you want to open. Either you have x-netbeans windows menu entries with your group1, group2, group3 ... groupx or you need to change this everytime you want to open netbeans with another group after you closes it.

If it works for you, fine and of course it is an idea. For other people it needs to be customizable as early as possible. For me should be the same.


> Well, if you set the dashboard to always show on startup, you've got easy one-click access to project groups or to switch between light and dark LAFs. If we expanded the widget list, you could have one just for group switching.

The dashboard comes after NetBeans opens the IDE and also the projects, it is in the progress of opening projects. This is IMHO to late.

**Comment by mbien** [comment]
> There is a related issue here that changing groups while projects are loading doesn't work. Making that work correctly and quickly might solve half the issue.

right. The automatic project open after launch will fail if the project group is switched while it is in progress. But once its done pg switching works fairly reliable in my experience. even when done quickly.

That is also the reason I pointed out `--close-group` exists, which will open the `none` group with nothing else loaded - avoiding the entire problem.

NB 30 can start in about 2.5s after initial cache creation (first start after each cluster activation takes longer) - on linux at least. So you should get to the project group button on the dashboard fairly quick. We might be able to put a combo box somewhere for the groups or make it easier to work with in other ways. I know there used to be a plugin which added a project group switcher.

I don't understand the requirement of it having to happen before nb starts. Implementing this before startup would be tricky (the startup sequence is already complex enough) and would likely also make it difficult to use FlatLaf for it.

The more maintainable way would be to add something similar to `--close-group` (e.g `--pick-group`) which would also start with group `none` but would open the group picker with the main window (`@OnShowing`). User safes a click.
